python/dlr/counter/counter_mgr.py

```diff
@@ -0,0 +1,144 @@
+import platform
+import atexit
+import json
+import hashlib
+import os
+import logging
+
+from .publisher import MsgPublisher
+from .system import Factory
+from . import config
+from .model_exec_counter import ModelExecCounter
+from .model_metric import ModelMetric
+
+
+def call_home(func):
+    def wrapped_call_home(*args, **kwargs):
+        call_counter = CallCounterMgr.get_instance()
+        if call_counter:
+            if func.__name__ == "init_call_home":
+                func(*args, **kwargs)
+                call_counter.runtime_loaded()
+            elif func.__name__ == "__init__":
+                func(*args, **kwargs)
+                obj = args[0]
+                call_counter.model_loaded(obj.get_model_name())
+            else:
+                res = func(*args, **kwargs)
+                obj = args[0]
+                call_counter.model_run(obj.get_model_name())
+                return res
+
+    return wrapped_call_home
+
+
+class CallCounterMgr(object):
+    RUNTIME_LOAD = 1
+    MODEL_LOAD = 2
+    MODEL_RUN = 3
+    _instance = None
+
+    @staticmethod
+    def get_instance():
+        """return single instance of class"""
+        if CallCounterMgr._instance is None:
+            if CallCounterMgr.is_feature_enabled():
+                print(config.CALL_HOME_USR_NOTIFICATION)
```

Comment:
I'd like this message to be part of setuptools entrypoint or post-installation. I'd like user being informed when they install DLR.

Reply:
If I understand this correctly, user needs to restart their application to load the static config file in order to shutdown the phone-home, correct?

Reply:
+ @mnuyens,
We have added this message in the setup script and will inform user before installing the DLR.
Yes, the user needs to restart their application after place the required configuration (to disable the feature).